src/lib_io.c

```diff
@@ -175,10 +180,18 @@ static int io_file_readlen(lua_State *L, FILE *fp, MSize m)
 {
   if (m) {
     char *buf = lj_str_needbuf(L, &G(L)->tmpbuf, m);
-    MSize n = (MSize)fread(buf, 1, m, fp);
-    setstrV(L, L->top++, lj_str_new(L, buf, (size_t)n));
+    MSize n;
+    for (;;) {
+      n = (MSize)fread(buf, 1, m, fp);
+      if (ferror(fp)) {
+        if (errno == EINTR) { clearerr(fp); continue; }
+      } else {
+        setstrV(L, L->top++, lj_str_new(L, buf, (size_t)n));
+      }
+      break;
+    }
     lj_gc_check(L);
-    return (n > 0 || m == 0);
```

Comment:
It seemed to me like the `m == 0` condition was superfluous, since I believe the above `if (m)` statement would never let this condition evaluate to true? If this change is undesired, I can revert it.